portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -64,4 +74,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String json = new Gson().toJson(commentList);
     response.getWriter().println(json);
   }
+
+  private String getUsername(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("user").setFilter(
+        new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+
+    if (entity == null) {
+      return null;
+    }
+
+    String nickname = (String) entity.getProperty("username");
+    return nickname;
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Query query = new Query("Comment");
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      List<Key> commentKeys = new ArrayList<>();
+      for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        Key taskEntityKey = KeyFactory.createKey("Comment", id);
+        commentKeys.add(taskEntityKey);
```

Comment:
Generally, I like to avoid extra variables. Up to you though.
```suggestion
        commentKeys.add(KeyFactory.createKey("Comment", id));
```

Reply:
Added inline